webapp/packages/plugin-authentication-administration/src/Administration/Users/UsersTable/useAdministrationUserFormState.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * CloudBeaver - Cloud Database Manager
+ * Copyright (C) 2020-2024 DBeaver Corp and others
+ *
+ * Licensed under the Apache License, Version 2.0.
+ * you may not use this file except in compliance with the License.
+ */
+import { observable } from 'mobx';
+import { useEffect } from 'react';
+
+import { useObservableRef } from '@cloudbeaver/core-blocks';
+import { IServiceProvider, useService } from '@cloudbeaver/core-di';
+
+import { AdministrationUserFormService } from '../UserForm/AdministrationUserFormService';
+import { AdministrationUserFormState } from '../UserForm/AdministrationUserFormState';
+
+export function useAdministrationUserFormState(id: string | null, configure?: (state: AdministrationUserFormState) => any) {
+  const service = useService(AdministrationUserFormService);
+  const serviceProvider = useService(IServiceProvider);
+  const ref = useObservableRef(
+    () => ({ formState: createFormState(id) }),
+    {
+      formState: observable.ref,
+    },
+    false,
+  );
+
+  function createFormState(id: string | null) {
+    const formState = new AdministrationUserFormState(serviceProvider, service, {
+      userId: id,
+    });
+    configure?.(formState);
+
+    return formState;
+  }
+
+  useEffect(() => {
+    if (ref.formState.id !== id) {
+      const formState = createFormState(id);
+      ref.formState = formState;
+    }
+  }, [id]);
```

Comment:
Seems a bit complicated. Can it be simplified like this?:

```ts
const ref = useRef<null | FormStateTypeHere>(null);

if(ref.current?.id !== id) {
  ref.current = new ...
}

return ref.current
```

the main difference:
form initialized and re-initialized same way (instead of init in render and re-initialization in the effect + re-render)
logic is straight forward
no need for mobx observable (cuz it will always produce correct state for provided id argument)

Reply:
yeah it can be simplified that way. thanks